packages/features/bookings/lib/handleNewBooking.ts

```diff
@@ -305,43 +312,151 @@ async function ensureAvailableUsers(
+        userFieldsResponses[field.name] = responses[field.name];
+      }
+    });
+    return {
+      ...reqBody,
+      name: responses.name,
+      email: responses.email,
+      guests: responses.guests ? responses.guests : [],
+      location: responses.location?.optionValue || responses.location?.value || "",
+      smsReminderNumber: responses.smsReminderNumber,
+      notes: responses.notes || "",
+      userFieldsResponses,
+      rescheduleReason: responses.rescheduleReason,
+    };
+  } else {
+    // Check if required custom inputs exist
+    handleCustomInputs(eventType.customInputs as EventTypeCustomInput[], reqBody.customInputs);
+
+    return {
+      ...reqBody,
+      name: reqBody.name,
+      email: reqBody.email,
+      guests: reqBody.guests,
+      location: reqBody.location || "",
+      smsReminderNumber: reqBody.smsReminderNumber,
+      notes: reqBody.notes,
+      rescheduleReason: reqBody.rescheduleReason,
+    };
+  }
+}
+
+function getCustomInputsResponses(
+  reqBody: {
+    responses?: Record<string, any>;
+    customInputs?: z.infer<typeof bookingCreateSchemaForApiOnly>["customInputs"];
+  },
+  eventTypeCustomInputs: Awaited<ReturnType<typeof getEventTypesFromDB>>["customInputs"]
+) {
+  const customInputsResponses = {} as NonNullable<CalendarEvent["customInputs"]>;
+  if ("customInputs" in reqBody) {
+    const reqCustomInputsResponses = reqBody.customInputs || [];
+    if (reqCustomInputsResponses?.length > 0) {
+      reqCustomInputsResponses.forEach(({ label, value }) => {
+        customInputsResponses[label] = value;
+      });
+    }
+  } else {
+    const responses = reqBody.responses || {};
+    // Backward Compatibility: Map new `responses` to old `customInputs` format so that webhooks can still receive same values.
+    for (const [fieldName, fieldValue] of Object.entries(responses)) {
+      const foundACustomInputForTheResponse = eventTypeCustomInputs.find(
+        (input) => slugify(input.label) === fieldName
+      );
+      if (foundACustomInputForTheResponse) {
+        customInputsResponses[foundACustomInputForTheResponse.label] = fieldValue;
+      }
+    }
+  }
+
+  return customInputsResponses;
+}
+
+async function handler(
+  req: NextApiRequest & { userId?: number | undefined },
+  {
+    isNotAnApiCall = false,
+  }: {
+    isNotAnApiCall?: boolean;
+  }
+) {
   const { userId } = req;
 
+  // handle dynamic user
+  let eventType =
+    !req.body.eventTypeId && !!req.body.eventTypeSlug
+      ? getDefaultEvent(req.body.eventTypeSlug)
+      : await getEventTypesFromDB(req.body.eventTypeId);
+
+  eventType = {
+    ...eventType,
+    bookingFields: getBookingFieldsWithSystemFields(eventType),
+  };
+
   const {
     recurringCount,
     allRecurringDates,
     currentRecurringIndex,
     noEmail,
-    eventTypeSlug,
     eventTypeId,
+    eventTypeSlug,
     hasHashedBookingLink,
     language,
     appsStatus: reqAppsStatus,
-    ...reqBody
```

Comment:
Isn't this the request body? Why the rename?

Reply:
Technically it is not reqBody now.
For an API request the body has `name` but for request from frontend the body has `responses.name` and this detail is normalized using `getBookingData` method so that `bookingData` has name in both the cases.